### PR TITLE
Improve embind when using c++17.

### PR DESCRIPTION
**Pull request by waywardmonkeys**
In C++17, the `noexcept` qualifier became part of the function
signature, so we must match it along with const / non-const.

### CODE REVIEW

**Comment by waywardmonkeys** [comment]
@chadaustin You might find this interesting!

**Comment by chadaustin** [comment]
😮 Seems reasonable to me, but is it possible to have the noexcept specialization defer to the non-noexcept specialization?

**Comment by waywardmonkeys** [comment]
Any further thoughts on this? I'd really like to see this land before the next emscipten release so that I can update to it more readily...

**Comment by kripken** [comment]
Looks like @chadaustin had a question?

Also maybe @jgravelle-google will have thoughts.

File: system/include/emscripten/bind.h [hunk 1]
```diff
@@ -1318,6 +1318,50 @@ namespace emscripten {
...
+
+        template<typename ClassType, typename ReturnType, typename... Args>
+        struct RegisterClassMethod<ReturnType (ClassType::*)(Args...) const noexcept> {
+
+            template <typename CT, typename... Policies>
+            static void invoke(const char* methodName,
+                               ReturnType (ClassType::*memberFunction)(Args...) const)  {
+                auto invoker = &MethodInvoker<decltype(memberFunction), ReturnType, const ClassType*, Args...>::invoke;
+
+                typename WithPolicies<Policies...>::template ArgTypeList<ReturnType, AllowedRawPointer<const ClassType>, Args...> args;
```
Comment (jgravelle-google) [inline system/include/emscripten/bind.h:1351]: Is this the only line of difference between the bodies of these two methods? That's really hard to spot in the middle of the rest of this duplicated code. Any way to compress these into one body?

Basically what @chadaustin said, yeah.

**Pull request closed without merging**